### ISSUE TITLE
Panic: `panic: unquote  err: invalid syntax`

### ISSUE THREAD

**Issue by justinsb** [hashicorp/terraform#6306]
Terraform panic-ed during a `terraform plan`

crash.log is as follows:
```
2016/04/22 11:27:57 [INFO] Terraform version: 0.6.14  c7e5b24531e73aae1eec2f7530d968ac8c69804c
2016/04/22 11:27:57 [DEBUG] Detected home directory from env var: /home/justinsb
2016/04/22 11:27:57 [DEBUG] Discovered plugin: atlas = /home/justinsb/bin/terraform/terraform-provider-atlas
2016/04/22 11:27:57 [DEBUG] Discovered plugin: aws = /home/justinsb/bin/terraform/terraform-provider-aws
2016/04/22 11:27:57 [DEBUG] Discovered plugin: azure = /home/justinsb/bin/terraform/terraform-provider-azure
2016/04/22 11:27:57 [DEBUG] Discovered plugin: azurerm = /home/justinsb/bin/terraform/terraform-provider-azurerm
2016/04/22 11:27:57 [DEBUG] Discovered plugin: chef = /home/justinsb/bin/terraform/terraform-provider-chef
2016/04/22 11:27:57 [DEBUG] Discovered plugin: clc = /home/justinsb/bin/terraform/terraform-provider-clc
2016/04/22 11:27:57 [DEBUG] Discovered plugin: cloudflare = /home/justinsb/bin/terraform/terraform-provider-cloudflare
2016/04/22 11:27:57 [DEBUG] Discovered plugin: cloudstack = /home/justinsb/bin/terraform/terraform-provider-cloudstack
2016/04/22 11:27:57 [DEBUG] Discovered plugin: consul = /home/justinsb/bin/terraform/terraform-provider-consul
2016/04/22 11:27:57 [DEBUG] Discovered plugin: datadog = /home/justinsb/bin/terraform/terraform-provider-datadog
2016/04/22 11:27:57 [DEBUG] Discovered plugin: digitalocean = /home/justinsb/bin/terraform/terraform-provider-digitalocean
2016/04/22 11:27:57 [DEBUG] Discovered plugin: dme = /home/justinsb/bin/terraform/terraform-provider-dme
2016/04/22 11:27:57 [DEBUG] Discovered plugin: dnsimple = /home/justinsb/bin/terraform/terraform-provider-dnsimple
2016/04/22 11:27:57 [DEBUG] Discovered plugin: docker = /home/justinsb/bin/terraform/terraform-provider-docker
2016/04/22 11:27:57 [DEBUG] Discovered plugin: dyn = /home/justinsb/bin/terraform/terraform-provider-dyn
2016/04/22 11:27:57 [DEBUG] Discovered plugin: github = /home/justinsb/bin/terraform/terraform-provider-github
2016/04/22 11:27:57 [DEBUG] Discovered plugin: google = /home/justinsb/bin/terraform/terraform-provider-google
2016/04/22 11:27:57 [DEBUG] Discovered plugin: heroku = /home/justinsb/bin/terraform/terraform-provider-heroku
2016/04/22 11:27:57 [DEBUG] Discovered plugin: influxdb = /home/justinsb/bin/terraform/terraform-provider-influxdb
2016/04/22 11:27:57 [DEBUG] Discovered plugin: mailgun = /home/justinsb/bin/terraform/terraform-provider-mailgun
2016/04/22 11:27:57 [DEBUG] Discovered plugin: mysql = /home/justinsb/bin/terraform/terraform-provider-mysql
2016/04/22 11:27:57 [DEBUG] Discovered plugin: null = /home/justinsb/bin/terraform/terraform-provider-null
2016/04/22 11:27:57 [DEBUG] Discovered plugin: openstack = /home/justinsb/bin/terraform/terraform-provider-openstack
2016/04/22 11:27:57 [DEBUG] Discovered plugin: packet = /home/justinsb/bin/terraform/terraform-provider-packet
2016/04/22 11:27:57 [DEBUG] Discovered plugin: postgresql = /home/justinsb/bin/terraform/terraform-provider-postgresql
2016/04/22 11:27:57 [DEBUG] Discovered plugin: powerdns = /home/justinsb/bin/terraform/terraform-provider-powerdns
2016/04/22 11:27:57 [DEBUG] Discovered plugin: rundeck = /home/justinsb/bin/terraform/terraform-provider-rundeck
2016/04/22 11:27:57 [DEBUG] Discovered plugin: statuscake = /home/justinsb/bin/terraform/terraform-provider-statuscake
2016/04/22 11:27:57 [DEBUG] Discovered plugin: template = /home/justinsb/bin/terraform/terraform-provider-template
2016/04/22 11:27:57 [DEBUG] Discovered plugin: terraform = /home/justinsb/bin/terraform/terraform-provider-terraform
2016/04/22 11:27:57 [DEBUG] Discovered plugin: tls = /home/justinsb/bin/terraform/terraform-provider-tls
2016/04/22 11:27:57 [DEBUG] Discovered plugin: triton = /home/justinsb/bin/terraform/terraform-provider-triton
2016/04/22 11:27:57 [DEBUG] Discovered plugin: ultradns = /home/justinsb/bin/terraform/terraform-provider-ultradns
2016/04/22 11:27:57 [DEBUG] Discovered plugin: vcd = /home/justinsb/bin/terraform/terraform-provider-vcd
2016/04/22 11:27:57 [DEBUG] Discovered plugin: vsphere = /home/justinsb/bin/terraform/terraform-provider-vsphere
2016/04/22 11:27:57 [DEBUG] Discovered plugin: chef = /home/justinsb/bin/terraform/terraform-provisioner-chef
2016/04/22 11:27:57 [DEBUG] Discovered plugin: file = /home/justinsb/bin/terraform/terraform-provisioner-file
2016/04/22 11:27:57 [DEBUG] Discovered plugin: local-exec = /home/justinsb/bin/terraform/terraform-provisioner-local-exec
2016/04/22 11:27:57 [DEBUG] Discovered plugin: remote-exec = /home/justinsb/bin/terraform/terraform-provisioner-remote-exec
2016/04/22 11:27:57 [DEBUG] Detected home directory from env var: /home/justinsb
2016/04/22 11:27:57 [DEBUG] Attempting to open CLI config file: /home/justinsb/.terraformrc
2016/04/22 11:27:57 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2016/04/22 11:27:57 [DEBUG] Detected home directory from env var: /home/justinsb
2016/04/22 11:27:57 [DEBUG] waiting for all plugin processes to complete...
panic: unquote  err: invalid syntax

goroutine 1 [running]:
panic(0xc2f980, 0xc82027f050)
	/opt/go/src/runtime/panic.go:464 +0x3e6
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/hcl/token.Token.Value(0x9, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/hcl/token/token.go:157 +0x785
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decodeString(0xc82014ea58, 0xc82023f660, 0x13, 0x7fcab6ca54c0, 0xc8202240a0, 0xc1ff40, 0xc82027eff0, 0x194, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:466 +0x246
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decode(0xc82014ea58, 0xc82023f660, 0x13, 0x7fcab6ca54c0, 0xc8202240a0, 0xc1ff40, 0xc82027eff0, 0x194, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:106 +0x936
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decodeInterface(0xc82014ea58, 0xc82023f660, 0x13, 0x7fcab6ca54c0, 0xc8202240a0, 0xc1ff40, 0xc82027eff0, 0x194, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:284 +0x50a
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decode(0xc82014ea58, 0xc82023f660, 0x13, 0x7fcab6ca54c0, 0xc8202240a0, 0xc1ff40, 0xc82027eff0, 0x194, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:98 +0x780
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decodeMap(0xc82014ea58, 0xc82027ef00, 0xd, 0x7fcab6ca5550, 0xc82015fd60, 0xc2b280, 0xc8200348e0, 0x195, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:378 +0x14c5
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decode(0xc82014ea58, 0xc82027ef00, 0xd, 0x7fcab6ca54f0, 0xc82021a360, 0xc2b280, 0xc8200348e0, 0x195, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:100 +0x7f4
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decodeSlice(0xc82014ea58, 0xc82027eea0, 0xa, 0x7fcab6ca54f0, 0xc82021a360, 0xc0bd20, 0xc82023f620, 0x97, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:446 +0x5af
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decode(0xc82014ea58, 0xc82027eea0, 0xa, 0x7fcab6ca54f0, 0xc82021a360, 0xc1ff40, 0xc82027eec0, 0x194, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:104 +0x8cc
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decodeMap(0xc82014ea58, 0xf98688, 0x4, 0x7fcab6ca5550, 0xc82015fcc0, 0xc2b280, 0xc8200348c8, 0x195, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:378 +0x14c5
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decode(0xc82014ea58, 0xf98688, 0x4, 0x7fcab6ca54f0, 0xc82001bf80, 0xc2b280, 0xc8200348c8, 0x195, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:100 +0x7f4
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.DecodeObject(0xbf6300, 0xc8200348c8, 0x7fcab6ca54f0, 0xc82001bf80, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:60 +0x25f
github.com/hashicorp/terraform/config.loadResourcesHcl(0xc82023ed20, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader_hcl.go:441 +0x939
github.com/hashicorp/terraform/config.(*hclConfigurable).Config(0xc82023ebe0, 0xf8fc18, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader_hcl.go:118 +0xd66
github.com/hashicorp/terraform/config.(*importTree).ConfigTree(0xc820214a40, 0x43, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/import_tree.go:87 +0x68
github.com/hashicorp/terraform/config.LoadFile(0xc82007f590, 0x43, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader.go:44 +0x80
github.com/hashicorp/terraform/config.LoadDir(0xc820010104, 0x37, 0x11, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader.go:91 +0x38c
github.com/hashicorp/terraform/config/module.NewTreeModule(0x0, 0x0, 0xc820010104, 0x37, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:40 +0x41
github.com/hashicorp/terraform/command.(*Meta).Context(0xc8201a60f0, 0xc820010104, 0x37, 0x1062e60, 0x11, 0x0, 0xa, 0x4, 0x7ffdc207a525, 0x0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta.go:148 +0x509
github.com/hashicorp/terraform/command.(*PlanCommand).Run(0xc8201a60f0, 0xc82000a1f0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/plan.go:66 +0x80d
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc8200da600, 0xc820125b90, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:153 +0x5ee
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:142 +0xb05
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:78 +0xa10
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:21 +0x18

**Comment by stack72** [comment]
Hi @justinsb 

apologies for this - can you please let us know the configuration (minus the secrets) you were using when this happened? This will help us try to recreate the error

Thanks

Paul

**Comment by justinsb** [comment]
I have tracked this down further.  I believe it happens when I put a null value essentially anywhere in the tf.json file.

**Comment by jen20** [comment]
@justinsb - do you have a config demonstrating this? Do you literally mean putting a `null` value as an object key (or similar) anywhere?

**Comment by mitchellh** [comment]
This is fixed in master, I don't know the exact issue number but I'm 100% certain I fixed this last week into 0.7.1.
